fpmsyncd/fpmsyncd.cpp

```diff
@@ -304,3 +327,30 @@ int main(int argc, char **argv)
 
     return 1;
 }
+
+void flushPipeline(RedisPipeline& pipeline, bool scheduled) {
+
+    size_t remaining = pipeline.size();
+
+    if (remaining == 0) {
+        SELECT_TIMEOUT = INFINITE;
+        return;
+    }
+
+    int idle = pipeline.getIdleTime();
+
+    // flush right away
+    if (remaining < SMALL_TRAFFIC || idle >= gFlushTimeout || idle <= 0 || scheduled) {
```

Comment:
Why flush with small traffic?
If there are lots of data, the pipeline will not flush until next timeout?

Reply:
This optimization aims to reduce pipeline flushes in the case of scaled traffic.

Hence for SMALL_TRAFFIC, do immediate flush to maintain the logic before optimization. So that it would not affect the performance when you want to dispatch several critical routes.

But since we skip several flushes, we need to set a timeout to make sure the routes would eventually get flushed, that's when ``` idle >= gFlushTimeout``` 

```idle <=0``` happens when the system clock is not steady, but since we use steady_clock it shouldn't happen.

Reply:
#Closed

Suggest add this info to code comments.